src/code/audio/audio_heap.c

```diff
@@ -1,111 +1,1576 @@
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B608.s")
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.sampleFontLoadStatus); i++) {
+        if (gAudioContext.sampleFontLoadStatus[i] != 5) {
+            gAudioContext.sampleFontLoadStatus[i] = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B640.s")
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.seqLoadStatus); i++) {
+        if (gAudioContext.seqLoadStatus[i] != 5) {
+            gAudioContext.seqLoadStatus[i] = 0;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B69C.s")
+void AudioHeap_DiscardFont(s32 fontId) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B6E8.s")
+    for (i = 0; i < gAudioContext.numNotes; i++) {
+        Note* note = &gAudioContext.notes[i];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B740.s")
+        if (note->playbackState.fontId == fontId) {
+            if (note->playbackState.unk_04 == 0 && note->playbackState.priority != 0) {
+                note->playbackState.parentLayer->enabled = false;
+                note->playbackState.parentLayer->finished = true;
+            }
+            AudioPlayback_NoteDisable(note);
+            AudioPlayback_AudioListRemove(&note->listItem);
+            AudioSeq_AudioListPushBack(&gAudioContext.noteFreeLists.disabled, &note->listItem);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B768.s")
+void AudioHeap_ReleaseNotesForFont(s32 fontId) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B77C.s")
+    for (i = 0; i < gAudioContext.numNotes; i++) {
+        Note* note = &gAudioContext.notes[i];
+        NotePlaybackState* playbackState = &note->playbackState;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B7AC.s")
+        if (playbackState->fontId == fontId) {
+            if (playbackState->priority != 0) {
+                playbackState->priority = 1;
+                playbackState->adsr.fadeOutVel = gAudioContext.audioBufferParameters.updatesPerFrameInv;
+                playbackState->adsr.action.s.release = true;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B7BC.s")
+void AudioHeap_DiscardSequence(s32 seqId) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B8FC.s")
+    for (i = 0; i < gAudioContext.audioBufferParameters.numSequencePlayers; i++) {
+        if (gAudioContext.seqPlayers[i].enabled && gAudioContext.seqPlayers[i].seqId == seqId) {
+            AudioSeq_SequencePlayerDisable(&gAudioContext.seqPlayers[i]);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_8018B95C.s")
+void* AudioHeap_WritebackDCache(void* mem, u32 size) {
+    Audio_WritebackDCache(mem, size);
+    if (mem) {}
+    return (void*)((u32)mem + 0x20000000);
```

Comment:
Do you know the meaning of `0x20000000`?
Also, this looks like a job for `uintptr_t`
```suggestion
    return (void*)((uintptr_t)mem + 0x20000000);
```